common/flatpak-dir.c

```diff
@@ -1866,6 +1871,12 @@ repo_pull_one_dir (OstreeRepo          *self,
         return flatpak_fail (error, "Update is older then current version");
     }
 
+  if ((flags & OSTREE_REPO_PULL_FLAGS_COMMIT_ONLY) == 0)
+    {
+      if (!flatpak_check_commit_content_perms (self, new_rev, cancellable, error))
```

Comment:
We have to do the same thing check in repo_pull_one_untrusted(), which is the codepath that the system-helper takes when pulling from the repo the user downloaded to the system repo.